assets/backport/github/github.go

```diff
@@ -0,0 +1,311 @@
+	// Create a new branch off of the target branch.
+	err := c.createBranchFrom(ctx, baseBranchName, newBranchName)
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+	fmt.Printf("Created a new branch: %s.\n", newBranchName)
+
+	commits, err := c.getPullRequestCommits(ctx, pullNumber)
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+	fmt.Printf("Found %v commits. \n", len(commits))
+
+	// Cherry pick commits.
+	err = c.cherryPickCommitsOnBranch(ctx, newBranchName, commits)
+	if err != nil {
+		return "", trace.Wrap(err)
+	}
+	return newBranchName, nil
+}
+
+// CreatePullRequest creates a pull request.
+func (c *Client) CreatePullRequest(ctx context.Context, baseBranch string, headBranch string, originalPrNumber int) error {
+	titleAndBody := fmt.Sprintf("Backport #%v to %s", originalPrNumber, baseBranch)
+	newPR := &go_github.NewPullRequest{
+		Title:               go_github.String(titleAndBody),
+		Head:                go_github.String(headBranch),
+		Base:                go_github.String(baseBranch),
+		Body:                go_github.String(titleAndBody),
+	}
+	_, _, err := c.Client.PullRequests.Create(ctx, c.Organization, c.Repository, newPR)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	return nil
+}
+
+// getPullRequestCommits gets the commits for a pull request.
+func (c *Client) getPullRequestCommits(ctx context.Context, number int) (commits []string, err error) {
+	var commitSHAs []string
+	opts := go_github.ListOptions{
+		Page:    0,
+		PerPage: perPage,
+	}
+	for {
+		currCommits, resp, err := c.Client.PullRequests.ListCommits(ctx,
+			c.Organization,
+			c.Repository,
+			number, &go_github.ListOptions{})
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
+
+		for _, commit := range currCommits {
+			commitSHAs = append(commitSHAs, commit.GetSHA())
+		}
+
+		if resp.NextPage == 0 {
+			break
+		}
+
+		opts.Page = resp.NextPage
+	}
+	return commitSHAs, nil
+}
+
+// cherryPickCommitsOnBranch cherry picks a list of commits on the given branch.
+func (c *Client) cherryPickCommitsOnBranch(ctx context.Context, branchName string, commits []string) error {
+	branch, _, err := c.Client.Repositories.GetBranch(ctx, c.Organization, c.Repository, branchName, true)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	// Get the branch's HEAD.
+	headCommit, _, err := c.Client.Git.GetCommit(ctx,
+		c.Organization,
+		c.Repository,
+		branch.GetCommit().GetSHA())
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	for i := 0; i < len(commits); i++ {
+		cherryCommit, _, err := c.Client.Git.GetCommit(ctx, c.Organization, c.Repository, commits[i])
+		if err != nil {
+			return trace.Wrap(err)
+		}
+		// Skip merge commits. The commit to cherry-pick MUST have only 1 parent.
+		if len(cherryCommit.Parents) != 1 {
+			continue
+		}
+		tree, sha, err := c.cherryPickCommit(ctx, branchName, cherryCommit, headCommit)
+		if err != nil {
+			return trace.Errorf("please manually delete branch %s: %v", branchName, err)
```

Comment:
Should this be a `trace.Wrap()`?

Also, at this point `branchName` is the the branch for  the new PR that is being cherry-picked onto, yes? Why can't the caller delete it on failure?